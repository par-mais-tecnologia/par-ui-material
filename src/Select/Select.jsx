import React, { PureComponent } from 'react'
import SelectMUI from '@material-ui/core/Select'
import FormControl from '@material-ui/core/FormControl/FormControl'
import InputLabel from '@material-ui/core/InputLabel'
import FormHelperText from '@material-ui/core/FormHelperText/FormHelperText'

class Select extends PureComponent {
  state = {
    open: false
  }

  validationInstance = this.props.validator ? this.props.validator.validator.register(
    this
  ) : false

  handleClose (event) {
    if (this.props.onClose) {
      return this.props.onClose(event)
    } else {
      this.setState({ open: false })
      return this.validationInstance ? this.validationInstance.validate(event) : false
    }
  }

  handleChange (event) {
    if (this.props.onChange) {
      this.props.onChange(event)
    }
  }

  handleOpen (event) {
    if (this.props.onFocus) {
      this.props.onFocus(event)
    } else {
      this.setState({ open: true })
    }
  }

  getStyle (props) {
    const { minWidth } = props
    return { minWidth: minWidth || '100%' }
  }

  render () {
    const inputProps = {
      onClose: this.handleClose.bind(this),
      onChange: this.handleChange.bind(this),
      onOpen: this.handleOpen.bind(this)
    }
    const { required, showLabel, id, label } = this.props
    if (this.validationInstance) {
      this.state.errors = {
        ...this.validationInstance.error
      }
    } else {
      this.state.errors = { hasError: false }
    }
    const proxyProps = { ...this.props }
    delete proxyProps.minWidth
    delete proxyProps.showError
    delete proxyProps.showLabel
    delete proxyProps.errorMessage

    return (
      <FormControl
        aria-describedby='select-form-control'
        required={required}
        error={this.state.errors.hasError}
        style={this.getStyle(this.props)}>
        {showLabel && label !== undefined
          ? <InputLabel htmlFor={id}>
            {label}
          </InputLabel> : ''
        }
        <SelectMUI
          {...proxyProps}
          {...inputProps}
          open={this.state.open}
          error={this.state.errors.hasError}
          inputProps={{
            name: label,
            id: { id }
          }} />
        {this.state.errors.hasError
          ? <FormHelperText style={{ marginTop: '2px', color: '#BA2B31' }}
            id={id}> {this.state.errors.errorMessage}
          </FormHelperText> : ''
        }
      </FormControl>
    )
  }
}

Select.defaultProps = {
  id: `Select ${Math.floor((Math.random() * 10) + 1)}`,
  showError: true,
  showLabel: true,
  required: false,
  errorMessage: '',
  error: false
}

export default Select
