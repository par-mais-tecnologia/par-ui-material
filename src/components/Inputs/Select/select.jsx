import React, { PureComponent } from 'react'
import SelectMUI from '@material-ui/core/Select'
import FormControl from '@material-ui/core/FormControl/FormControl'
import InputLabel from '@material-ui/core/InputLabel'
import FormHelperText from '@material-ui/core/FormHelperText/FormHelperText'

class Select extends PureComponent {
  render () {
    const { required, showLabel, showError, errorMessage, id, label } = this.props

    return (
      <FormControl
        aria-describedby='select-form-control'
        required={required}>
        {showLabel ? <InputLabel htmlFor={id}>{label}</InputLabel> : ''}
        <SelectMUI
          {...this.props}
          inputProps={{
            name: { label },
            id: { id }
          }} />
        {showError ? <FormHelperText id='email-form-control'> {errorMessage} </FormHelperText> : ''}
      </FormControl>
    )
  }
}

Select.defaultProps = {
  id: `Select ${Math.floor((Math.random() * 10) + 1)}`,
  showError: true,
  showLabel: true,
  required: false,
  errorMessage: '',
  hasError: false
}

export default Select
