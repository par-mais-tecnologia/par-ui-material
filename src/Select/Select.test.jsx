import React from 'react'
import { configure, shallow } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import Select from './'

configure({ adapter: new Adapter() })

export const test = describe('Select', () => {
  it('shallow Select', () => {
    const wrapper = shallow(
      <Select id={'1'} />
    )
    expect(wrapper).toBeTruthy()
  })

  it('shallow Select match Snapshot', () => {
    const wrapper = shallow(
      <Select id={'1'} />
    )
    expect(wrapper).toMatchSnapshot()
  })
})
