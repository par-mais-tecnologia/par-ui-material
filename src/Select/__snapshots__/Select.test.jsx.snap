// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Select shallow Select match Snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Select
    error={false}
    errorMessage=""
    id="1"
    required={false}
    showError={true}
    showLabel={true}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "aria-describedby": "select-form-control",
      "children": Array [
        <WithStyles(InputLabel)
          htmlFor="1"
        />,
        <WithStyles(Select)
          error={false}
          id="1"
          inputProps={
            Object {
              "id": Object {
                "id": "1",
              },
              "name": undefined,
            }
          }
          onChange={[Function]}
          onClose={[Function]}
          onOpen={[Function]}
          open={false}
          required={false}
        />,
        "",
      ],
      "error": false,
      "required": false,
      "style": Object {
        "minWidth": "100%",
      },
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": undefined,
          "htmlFor": "1",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "error": false,
          "id": "1",
          "inputProps": Object {
            "id": Object {
              "id": "1",
            },
            "name": undefined,
          },
          "onChange": [Function],
          "onClose": [Function],
          "onOpen": [Function],
          "open": false,
          "required": false,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "",
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "aria-describedby": "select-form-control",
        "children": Array [
          <WithStyles(InputLabel)
            htmlFor="1"
          />,
          <WithStyles(Select)
            error={false}
            id="1"
            inputProps={
              Object {
                "id": Object {
                  "id": "1",
                },
                "name": undefined,
              }
            }
            onChange={[Function]}
            onClose={[Function]}
            onOpen={[Function]}
            open={false}
            required={false}
          />,
          "",
        ],
        "error": false,
        "required": false,
        "style": Object {
          "minWidth": "100%",
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": undefined,
            "htmlFor": "1",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "error": false,
            "id": "1",
            "inputProps": Object {
              "id": Object {
                "id": "1",
              },
              "name": undefined,
            },
            "onChange": [Function],
            "onClose": [Function],
            "onOpen": [Function],
            "open": false,
            "required": false,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "",
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
